### PR TITLE
Add start-up functionality to check if Qt application settings need to be migrated.

### LINKED ISSUE #332: bitcoind can't find the datadir on windows
This is on the current dev branch...when we launch bitcoind and we have previously bitcoin it should find our datadir if it's not in the default location.  There should be a reg setting that it's referencing from previous installations however how it doesn't work anymore since the latest merges to dev.  Not sure which merge was the problem.

### PR DESCRIPTION
**Pull request by Justaphf**
Fixes #332 

75c7b7f caused an issue where all previously saved application settings were "lost" due to the expected registry path on Windows being changed.  The registry path is generated using the value of `QAPP_ORG_NAME` so bitcoin-qt is now looking for 'BitcoinUnlimited' instead of 'Bitcoin'.

Marked as WIP for the following reasons:
1. I have only tested on Windows, still need to test on other OS which use file based settings instead of registry.
2. This handles the case where there are no alternate settings to migrate, or the user has not made any changes to the settings since upgrading past v1.0.1.0.  However, I'm not sure the best way to handle the case where there are old settings under the 'Bitcoin' registry key which can be migrated AND the user has made additional changes to settings after transitioning to the 'BitcoinUnlimited' registry key.  The current implementation will just overwrite the values currently in 'BitcoinUnlimited' with the values in 'Bitcoin'.

### CODE REVIEW

**Comment by Justaphf** [comment]
Verified on a Debian Jessie machine and file based migration also works correctly.

For testing purposes, the config file generated by Qt is located in `~/.config/$QAPP_ORG_NAME/Bitcoin-Qt.conf`, in lieu of the Windows Registry.

Moving from WIP pending review and suggestions against point 2 in my previous comment.

**Comment by AllanDoensen** [comment]
I am looking at testing this. So just a clarification: I would see this issue if on windows I move from running the 'release' to the "dev' branch of the current BU codebase?

**Comment by Justaphf** [comment]
@AllanDoensen thanks for reviewing.  At this point the code that causes the expected registry to change is in release as well (7395d35).  You would need to install v1.0.0.0 or older to have the old registry, then upgrade to v1.0.1.0 or newer to cause the issue, then use this PR to trigger the migration from old registry to new registry.

The original issue was that if you had a custom data directory, then upgraded to v1.0.1.0 or later the client starts downloading the block chain from scratch in the default directory, but after investigating I found that due to the new registry location it basically reset all user customizable parameters stored in the registry to defaults, not just data directory.

**Comment by Justaphf** [comment]
Another note, this PR just deals with migration after the fact, I did not check/change anything with the windows installer itself.

File: src/qt/bitcoin.cpp [hunk 1]
```diff
@@ -514,6 +514,127 @@ WId BitcoinApplication::getMainWinId() const
...
+* Create a backup of Qt App Settings
+* Backup will only be performed if there are settings to backup and the backup
+* location is writable.
+* @param[in] source      App settings to be backed up
+* @param[in] backupName  Backup location name
+* @return true if backup was successful or there were no settings to backup, otherwise false
+*/
+bool BackupQtAppSettings(const QSettings &source, const QString &backupName)
+{
+    //parameter saftey check
```
Comment (gandrewstone) [inline src/qt/bitcoin.cpp:552]: spelling: safety

**Comment by gandrewstone** [comment]
I tested this on ubuntu by moving from core 14 to this branch of BU

**Comment by sickpig** [comment]
@Justaphf @gandrewstone does this PR need to be backported to `release` branch?

**Pull request merged**